code/datums/diseases/advance/symptoms/sensory.dm

```diff
@@ -30,21 +30,33 @@ Bonus
 		var/mob/living/M = A.affected_mob
 		switch(A.stage)
 			if(2)
-				if(M.reagents.get_reagent_amount("oculine")<10)
+				if(M.reagents.get_reagent_amount("oculine") < 10)
 					M.reagents.add_reagent("oculine", 10)
 					to_chat(M, "<span class='notice'>Your hearing feels clearer and crisp.</span>")
 			if(3)
-				if(M.reagents.get_reagent_amount("antihol") < 10 && M.reagents.get_reagent_amount("oculine") < 10 )
-					M.reagents.add_reagent_list(list("antihol"=10, "oculine"=10))
+				if(M.reagents.get_reagent_amount("antihol") < 10)
+					M.reagents.add_reagent("antihol" = 10)
 					to_chat(M, "<span class='notice'>You feel sober.</span>")
+				if(M.reagents.get_reagent_amount("oculine") < 10)
+					M.reagents.add_reagent("oculine", 10)
 			if(4)
-				if(M.reagents.get_reagent_amount("antihol") < 10 && M.reagents.get_reagent_amount("oculine") < 10 && M.reagents.get_reagent_amount("synaphydramine") < 10)
-					M.reagents.add_reagent_list(list("antihol"=10, "oculine"=10, "synaphydramine"=5))
+				if(M.reagents.get_reagent_amount("synaphydramine") < 10)
+					M.reagents.add_reagent("synaphydramine" = 5)
 					to_chat(M, "<span class='notice'>You feel focused.</span>")
+				if(M.reagents.get_reagent_amount("antihol") < 10)
+					M.reagents.add_reagent("antihol" = 10)
+				if(M.reagents.get_reagent_amount("oculine") < 10)
+					M.reagents.add_reagent("oculine", 10)
 			if(5)
-				if(M.reagents.get_reagent_amount("antihol") < 10 && M.reagents.get_reagent_amount("oculine") < 10 && M.reagents.get_reagent_amount("synaphydramine") < 10 && M.reagents.get_reagent_amount("mannitol") < 10)
-					M.reagents.add_reagent_list(list("mannitol"=10, "antihol"=10, "oculine"=10, "synaphydramine"=10))
+				if(M.reagents.get_reagent_amount("mannitol") < 10)
+					M.reagents.add_reagent("mannitol" = 10)
 					to_chat(M, "<span class='notice'>Your mind feels relaxed.</span>")
```

Comment:
Maybe make the proc return a bool so you can show a message like this if it's applied.